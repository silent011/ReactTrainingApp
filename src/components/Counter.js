import React from 'react'

let Counter = ({count, inc, message}) => {
    console.log('count',count)
   return <div>
        <h2>Counter page:</h2>
        <h2 id="count-value">Count: {count}</h2>
        <button onClick={inc}>Increment</button>   
        <h3>{message}</h3>   
    </div>
}

export default Counter