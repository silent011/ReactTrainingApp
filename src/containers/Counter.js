import React from 'react'
import CounterView from '../components/Counter'
import { connect } from 'react-redux'
import { incrementByOne } from '../actions/count-actions'

let custom = {
    message: 'Hello from ownProps2'
}


let mapStateToProps = (state, ownProps) => {
    console.log('own props', ownProps)
    return {
        count: Object.assign({}, state.count),
        message: custom.message
    }
}

let mapDispatchToProps = dispatch => {
    return {
        inc: () => { dispatch(incrementByOne()) }
    }
}



let Counter = connect(mapStateToProps, mapDispatchToProps)(CounterView)

export default Counter