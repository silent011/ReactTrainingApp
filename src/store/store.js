import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import countReducer from '../reducers/count-reducer'
import thunk from 'redux-thunk'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(
    combineReducers({count: countReducer}),
    composeEnhancers(applyMiddleware(thunk))
)

export default store