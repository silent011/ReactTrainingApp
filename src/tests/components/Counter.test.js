import React from 'react'
import { shallow } from 'enzyme'
import { mount } from 'enzyme'
import Counter from '../../components/Counter'
import CounterContainer from '../../containers/Counter'
import state from '../fixtures/count'
import { Provider } from 'react-redux'
import store from '../../store/store'

test('should render the same initial ui', () => {
    const wrapper = shallow(<Counter {...state}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should give a count value equal to the input count', () => {
    const wrapper = shallow(<Counter {...state} count={1}/>)
    expect(wrapper.find('#count-value').text()).toBe('Count: 1')
})

test('should increment by 1 on click', () => {
    const wrapper = mount(<CounterContainer store={store} />)
    wrapper.find('button').simulate('click')
    expect(wrapper.find('#count-value').text()).toBe('Count: 1')
})