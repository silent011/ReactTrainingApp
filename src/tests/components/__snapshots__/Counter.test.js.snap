// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the same initial ui 1`] = `
<div>
  <h2>
    Counter page:
  </h2>
  <h2
    id="count-value"
  >
    Count: 
    0
  </h2>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <h3>
    Hello from ownProps
  </h3>
</div>
`;
